src/urllib3/fields.py

```diff
@@ -231,28 +272,34 @@ def render_headers(self):
         return "\r\n".join(lines)
 
     def make_multipart(
-        self, content_disposition=None, content_type=None, content_location=None
-    ):
+        self,
+        content_disposition: Optional[str] = None,
+        content_type: Optional[str] = None,
+        content_location: Optional[str] = None,
+    ) -> None:
         """
         Makes this request field into a multipart request field.
 
         This method overrides "Content-Disposition", "Content-Type" and
         "Content-Location" headers to the request parameter.
 
+        :param content_disposition:
+            The 'Content-Disposition' of the request body. Defaults to 'form-data'
         :param content_type:
             The 'Content-Type' of the request body.
         :param content_location:
             The 'Content-Location' of the request body.
 
         """
-        self.headers["Content-Disposition"] = content_disposition or "form-data"
```

Comment:
Would it be better to encode the default value in the kwarg? This would avoid using `Optional` and would be consistent with `guess_content_type`.

I understand that we want to just add types and not change the code, but this type of small change is worth it in my opinion as it results in better types.

Reply:
I like this idea! We can keep functional compatibility + stay defensive by keeping the `... or "form-data"` but still recommend users only pass `str`.

Reply:
Hmm, I hadn't thought about keeping the default value in the code. It's s good point. That would mean keeping two copies of that default value, and they'll go out of sync eventually.

I'm now only in favor of this idea if we say the argument isn't Optional and accept that passing None will now break. Not sure if this is acceptable or not.

Reply:
Probably not acceptable. I added a commit with the other accepted change, and will merge if tests pass.